sched/worker/workers/polling.go

```diff
@@ -41,29 +41,38 @@ type PollingWorker struct {
 	timeout        time.Duration
 }
 
-func (r *PollingWorker) RunAndWait(task sched.TaskDefinition) (runner.ProcessStatus, error) {
-	// schedule the task
-	status, err := r.runner.Run(&task.Command)
-	if err != nil {
-		return status, err
-	}
+func (r *PollingWorker) Start(task sched.TaskDefinition) (runner.ProcessStatus, error) {
+	return r.runner.Run(&task.Command)
+}
 
-	timeSpent := 0 * time.Second
-	id := status.RunId
+func (r *PollingWorker) Status(runId runner.RunId) (runner.ProcessStatus, error) {
+	return r.runner.Status(runId)
+}
 
+func (r *PollingWorker) Wait(runId runner.RunId) (status runner.ProcessStatus, err error) {
 	// Periodically check in with the worker to get a task update
+	timeSpent := 0 * time.Second
 	for !r.enforceTimeout || (r.enforceTimeout && timeSpent < r.timeout) {
 		time.Sleep(r.pollingPeriod)
 		timeSpent += r.pollingPeriod
 
-		status, err = r.runner.Status(id)
+		status, err = r.runner.Status(runId)
 		if err != nil || status.State.IsDone() {
-			return status, err
+			return
```

Comment:
i like the named return values here, but it's a little weird with Start, Status, and RunAndWait using explicit returns.